src/coord/src/persistcfg.rs

```diff
@@ -25,12 +28,39 @@ use persist::indexed::runtime::{
 };
 use uuid::Uuid;
 
+#[derive(Clone, Debug)]
+pub enum PersistStorage {
+    File(PersistFileStorage),
+    S3(PersistS3Storage),
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct PersistFileStorage {
+    /// A directory under which larger batches of indexed data are stored.
+    pub blob_path: PathBuf,
+}
+
+#[derive(Clone, Debug)]
+pub struct PersistS3Storage {
+    /// An S3 bucket in which larger batches of indexed data are stored.
+    pub bucket: String,
+    /// A prefix prepended for all S3 object keys used by this storage. Empty is
+    /// fine.
+    pub prefix: String,
+}
+
 /// Configuration of the persistence runtime and features.
 #[derive(Clone, Debug)]
 pub struct PersistConfig {
-    /// A directory under which larger batches of indexed data are stored. This
-    /// will eventually be S3 for Cloud.
-    pub blob_path: PathBuf,
+    /// A runtime used for IO and cpu heavy work. The None case should only be
+    /// used for [PersistConfig::disabled] (which is an awkward artifact of us
+    /// currently initializing persistence in the Catalog startup).
+    //
+    // WIP should we plumb this through catalog's Config to the
```

Comment:
@benesch looking for your thoughts here

Reply:
If this is simpler I think it's fine! The catalog/coord startup is due for a refactor, but I don't have more structured thoughts right now.